ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -149,6 +165,62 @@ func (s *indexerStreamingService) publishTxn(ctx context.Context, req abci.Reque
 	return nil
 }
 
+// adjustAmountBySpreadFactor adjusts the amount by the spread factor.
+// This is done to adjust the amount of tokens in the token_swapped event by the spread factor,
+// as the amount in the event is the amount AFTER the spread factor is applied.
+// therefore, we need to adjust the amount by the spread factor to get the amount BEFORE the spread factor is applied.
+// NOTE: This applies to CL pools only
+func (s *indexerStreamingService) adjustTokenInAmountBySpreadFactor(ctx context.Context, event *abci.Event) error {
```

Comment:
Thanks. Could you please also add a happy path test before merge based on a sample event constructed from Mintscan?

Reply:
Thanks. I've created a [draft PR](https://github.com/osmosis-labs/osmosis/pull/8544) for indexer test cases as well as Linear issues [IND-91](https://linear.app/osmosis/issue/IND-91/add-unit-test-cases-for-adjusttokeninamountbyspreadfactor-in-ingester) and [IND-90](https://linear.app/osmosis/issue/IND-90/add-unit-test-cases-for-addtokenliquidity-in-ingester) to track them. 

Will request a session with you tomorrow about passing an event object to the test methods. 

Since staging and production environment are dependent on this fix and we want to receive Dexscreener feedback asap if any, if you don't mind and okay with the above arrangement, I'm going to merge this PR while I'm working on the test cases. Thanks!

Reply:
Merged it so that we can start rebuilding staging & production indexer env from scratch by EU morning, and then hand over the env to Dexscreener to next iteration for review.